extensions/wikia/WikiaApi/WikiaApiQueryAllUsers.php

```diff
@@ -170,89 +158,89 @@ private function global_users() {
+		$res = $this->select( __METHOD__ );
 
 		$data = array ();
 		$count = 0;
 		$lastUserData = false;
 		$lastUser = false;
 		$result = $this->getResult();
 
-		while (true) {
+		while ( true ) {
 
-			$row = $db->fetchObject($res);
+			$row = $db->fetchObject( $res );
 			$count++;
 
-			if (!$row || $lastUser !== $row->user_name) {
+			if ( !$row || $lastUser !== $row->user_name ) {
 				// Save the last pass's user data
-				if (is_array($lastUserData)) {
-					$fit = $result->addValue( array('query', $this->getModuleName()), null, $lastUserData );
-					if(!$fit) {
-						$this->setContinueEnumParameter( 'from', $this->keyToTitle($lastUserData['name']) );
+				if ( is_array( $lastUserData ) ) {
+					$fit = $result->addValue( array( 'query', $this->getModuleName() ), null, $lastUserData );
+					if ( !$fit ) {
+						$this->setContinueEnumParameter( 'from', $this->keyToTitle( $lastUserData['name'] ) );
 						break;
 					}
 				}
 
 				// No more rows left
-				if (!$row) {
+				if ( !$row ) {
 					break;
 				}
 
-				if ($count > $limit) {
+				if ( $count > $limit ) {
 					// We've reached the one extra which shows that there are additional pages to be had. Stop here...
-					$this->setContinueEnumParameter('from', $this->keyToTitle($row->user_name));
+					$this->setContinueEnumParameter( 'from', $this->keyToTitle( $row->user_name ) );
 					break;
 				}
 
 				// Record new user's data
 				$lastUser = $row->user_name;
 				$lastUserData = array( 'name' => $lastUser, 'id' => $row->user_id );
-				if ($this->fld_blockinfo) {
+				if ( $this->fld_blockinfo ) {
 					$lastUserData['blockedby'] = $row->blocker_name;
 					$lastUserData['blockreason'] = $row->ipb_reason;
 				}
-				if ($this->fld_editcount) {
-					$lastUserData['editcount'] = intval($row->user_editcount);
+				if ( $this->fld_editcount ) {
+					$lastUserData['editcount'] = intval( $row->user_editcount );
 				}
-				if ($this->fld_registration) {
-					$lastUserData['registration'] = wfTimestamp(TS_ISO_8601, $row->user_registration);
+				if ( $this->fld_registration ) {
+					$lastUserData['registration'] = wfTimestamp( TS_ISO_8601, $row->user_registration );
 				}
 
-			// Add user's group info
+				// Add user's group info
 				if ( $this->fld_groups ) {
-					$lastUserData['groups'] = array_key_exists($row->user_id, $user_groups) ? $user_groups[$row->user_id] : array();
-					$result->setIndexedTagName($lastUserData['groups'], 'g');
+					$lastUserData['groups'] = ( !empty( $users ) && array_key_exists( $row->user_id, $users ) )
+						? explode( ';', $users[$row->user_id] )
+						: array();
+					$result->setIndexedTagName( $lastUserData['groups'], 'g' );
 				}
 			}
 
 		}
 
-		$db->freeResult($res);
+		$db->freeResult( $res );
 
-		$result->setIndexedTagName_internal(array('query', $this->getModuleName()), 'u');
+		$result->setIndexedTagName_internal( array( 'query', $this->getModuleName() ), 'u' );
 	}
 
 	private function local_users() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "local_users" is 24 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)